walk-in-interview/src/test/java/com/google/job/data/JobsDatabaseTest.java

```diff
@@ -209,6 +208,56 @@ public void fetchJob_NormalInput_success() throws ExecutionException, Interrupte
         assertEquals(job, actualJob);
     }
 
+    @Test
+    public void isJobIdExist_NormalInput_true() throws ExecutionException, InterruptedException {
+        // Arrange.
+        Job job = new Job();
+        Future<DocumentReference> addedJobFuture = firestore.collection(TEST_JOB_COLLECTION).add(job);
+
+        DocumentReference documentReference = addedJobFuture.get();
+        // Asynchronously retrieve the document.
+        ApiFuture<DocumentSnapshot> future = documentReference.get();
+
+        // future.get() blocks on response.
+        DocumentSnapshot document = future.get();
+        String jobId = document.getId();
+
+        // Act.
+        boolean isExist = JobsDatabase.isJobIdExist(jobId);
+
+        // Assert.
+        assertTrue(isExist);
+    }
+
+    @Test
+    public void isJobIdExist_EmptyJobId_illegalArgumentException() throws ExecutionException, InterruptedException {
+        // Arrange.
+        Job job = new Job();
+        firestore.collection(TEST_JOB_COLLECTION).add(job);
+
+        try {
+            // Act.
+            boolean isExist = JobsDatabase.isJobIdExist("");
+            fail();
+        } catch (IllegalArgumentException e) {
+            // Assert.
+            assertEquals("Empty Job Id", e.getMessage());
+        }
+    }
+
+    @Test
+    public void isJobIdExist_InvalidJobId_false() throws ExecutionException, InterruptedException {
+        // Arrange.
+        Job job = new Job();
+        firestore.collection(TEST_JOB_COLLECTION).add(job);
+
+        // Act.
+        boolean isExist = JobsDatabase.isJobIdExist("dummy");
```

Comment:
how do we know that this jobId is invalid?

Reply:
cloud firestore id is not that short